nucypher/blockchain/eth/interfaces.py

```diff
@@ -418,7 +418,7 @@ def call_backend_sign(self, account: str, message: bytes) -> str:
             address = to_canonical_address(account)
             sig_key = provider.ethereum_tester.backend._key_lookup[address]
             signed_message = sig_key.sign_msg(message)
-            return signed_message.to_hex()
+            return signed_message
```

Comment:
This breaks parity with the rest of the web3 API.

So, if we're running a test, this API is called, which returns a string. In most cases, this is what we should expect when calling some node to sign something for us, not the object.

Reply:
This is why I hate this code and want a better signing solution from web3py here.

Reply:
I... don't see the problem.

Reply:
In one case (running tests), we return a `Signature`. In another, we return a string.

This is not good API design for signature, imho. They should both return an object (which is problematic), or they should both return a string.

Reply:
Ahh, I see.  So `w3.eth.sign` returns a string?

In any case, remember this is a [WIP] PR - I haven't even looked at the `else` clause in this function yet.